diff --git a/lib/python/Plugins/SystemPlugins/Videomode/VideoHardware.py b/lib/python/Plugins/SystemPlugins/Videomode/VideoHardware.py
index 78f2bc756..3a46a9a13 100644
--- a/lib/python/Plugins/SystemPlugins/Videomode/VideoHardware.py
+++ b/lib/python/Plugins/SystemPlugins/Videomode/VideoHardware.py
@@ -50,9 +50,9 @@ class VideoHardware:
 								"auto":		{ 50: "2160p25", 60: "2160p30", 24: "2160p24" } }
 
 	rates["2160p"] =		{ "50Hz":	{ 50: "2160p50" },
-								"60Hz":		{ 60: "2160p" },
-								"multi":	{ 50: "2160p50", 60: "2160p" }, 
-								"auto":		{ 50: "2160p50", 60: "2160p", 24: "2160p24" }}
+								"60Hz":		{ 60: "2160p60" },
+								"multi":	{ 50: "2160p50", 60: "2160p60" }, 
+								"auto":		{ 50: "2160p50", 60: "2160p60", 24: "2160p24" }}
 
 	rates["PC"] = {
 		"1024x768": { 60: "1024x768" }, # not possible on DM7025
@@ -77,7 +77,7 @@ class VideoHardware:
 	if SystemInfo["HasYPbPr"]:
 		modes["YPbPr"] = ["720p", "1080i", "576p", "480p", "576i", "480i"]
 	if SystemInfo["Has2160p"]:
-		modes["DVI"] = ["720p", "1080p", "2160p", "1080i", "576p", "480p", "576i", "480i"]
+		modes["DVI"] = ["720p", "1080p", "2160p", "2160p60","1080i", "576p", "480p", "576i", "480i"]
 	else:
 		modes["DVI"] = ["720p", "1080p", "2160p", "2160p30", "1080i", "576p", "480p", "576i", "480i"]
 	modes["DVI-PC"] = ["PC"]
@@ -117,7 +117,7 @@ class VideoHardware:
 		self.current_port = None
 
 		self.readAvailableModes()
-		self.widescreen_modes = set(["720p", "1080i", "1080p", "2160p", "2160p30"]).intersection(*[self.modes_available])
+		self.widescreen_modes = set(["720p", "1080i", "1080p", "2160p", "2160p30", "2160p60"]).intersection(*[self.modes_available])
 
 		if "DVI-PC" in self.modes and not self.getModeList("DVI-PC"):
 			print "[VideoHardware] remove DVI-PC because of not existing modes"
diff --git a/lib/python/Plugins/SystemPlugins/Videomode/VideoWizard.py b/lib/python/Plugins/SystemPlugins/Videomode/VideoWizard.py
index 7c62b01ae..7f7944d51 100644
--- a/lib/python/Plugins/SystemPlugins/Videomode/VideoWizard.py
+++ b/lib/python/Plugins/SystemPlugins/Videomode/VideoWizard.py
@@ -134,7 +134,7 @@ class VideoWizard(WizardLanguage, Rc):
 	def modeSelect(self, mode):
 		ratesList = self.listRates(mode)
 		print "[VideoWizard] ratesList:", ratesList
-		if self.port == "DVI" and mode in ("720p", "1080i", "1080p", "2160p", "2160p30"):
+		if self.port == "DVI" and mode in ("720p", "1080i", "1080p", "2160p", "2160p30", "2160p60"):
 			if SystemInfo["Has24hz"]:
 				self.rate = "auto"
 				self.hw.setMode(port = self.port, mode = mode, rate = "auto")
